### PR TITLE
[ROCM] Modify transcoding: absolute path ->relative path

### LINKED ISSUE #91797: When the rocm environment is compiled using torch transcoding, packaging errors will occur.
### 🐛 Describe the bug

When the rocm environment is compiled using torch transcoding, packaging errors will occur.
The error message is as follows:
image

A recurring test project.
[rocm_test.zip](https://github.com/pytorch/pytorch/files/10358802/rocm_test.zip)

### Versions

torch1.12

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport

### PR DESCRIPTION
**Pull request by kahakuka**
Fixes https://github.com/pytorch/pytorch/issues/91797
This PR compiles the transcoded file with a relative path to ensure that the written transcoded file is written to SOURCE.txt as a relative path. Ensure successful packaging.

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport

### CODE REVIEW

**Comment by ezyang** [comment]
Does this fix a reported issue? Can you write a test?

**Comment by xiabo123** [comment]
@ezyang 
You can refer to the issuse below, which provides testing projects.
https://github.com/pytorch/pytorch/issues/91797

**Comment by ezyang** [comment]
ok batting this over to the rocm peeps

**Comment by jithunnair-amd** [comment]
We are looking into the changes in this PR.

File: torch/utils/cpp_extension.py
Comment (jithunnair-amd) [inline torch/utils/cpp_extension.py:1092]: Suggest to remove the above snippet and use the snippet below for minimal changes
```suggestion
```

```
        hipified_sources = set()
        for source in sources:
            s_abs = os.path.abspath(source)
            hipified_s_abs = (hipify_result[s_abs]["hipified_path"] if (s_abs in hipify_result and
                             hipify_result[s_abs]["hipified_path"] is not None) else s_abs)
            # setup() arguments must *always* be /-separated paths relative to the setup.py directory, 
            # *never* absolute paths
            hipified_sources.add(os.path.relpath(hipified_s_abs, build_dir))

        sources = list(hipified_sources)
```

**Review by jithunnair-amd — CHANGES_REQUESTED** [review]
@xiabo123 Thanks for your PR. I would recommend using the suggested snippet since it's more concise and also doesn't have the unnecessary "." in the beginning of the path.

The PR intent is correct, it resolves the error that shows up when using `use_package_data=True` in setup.py with hipified_sources.

I tested the suggested changes with apex and torchvision and the SOURCES.txt files were equivalent. I also tested them with the provided testcase.

cc @ezyang @jeffdaily

File: torch/utils/cpp_extension.py
Comment (xiabo123) [reply torch/utils/cpp_extension.py:1092]: @jithunnair-amd Thank you for your suggestion. I'll revise it here.

**Review by jithunnair-amd — APPROVED** [review]
LGTM. This PR only modifies the paths being passed to the Extension class from absolute to relative. This doesn't affect the paths being passed in to hipify.

**Comment by jeffdaily** [comment]
@ezyang we approve.  Waiting for maintainer approval before we can merge.

**Review by ezyang — APPROVED** [review]

**Comment by xiabo123** [comment]
@jithunnair-amd @jeffdaily @ezyang 
Hello, thank you for your approval of the modification. I think the failure of the additional jobs([trunk](https://github.com/pytorch/pytorch/actions/runs/3888567777) ,[trunk / linux-focal-rocm5.3-py3.8 / test (default, 2, 2, linux.rocm.gpu)](https://github.com/pytorch/pytorch/actions/runs/3888567777/jobs/6636262099)) is the reason for this modification. May I trouble you for a look?
I think the final result of the build log is OK. When generating the report, timed out.The final log screenshot is as follows, which can also be downloaded on pr.
image

**Pull request closed without merging**
